Makefile

```diff
@@ -204,16 +207,32 @@ $(BINODIR)/%.bin: $(IMAGEODIR)/%.out
 # Should be done in top-level makefile only
 #
 
-all:	sdk_pruned pre_build .subdirs $(OBJS) $(OLIBS) $(OIMAGES) $(OBINS) $(SPECIAL_MKTARGETS)
+all: toolchain sdk_pruned pre_build .subdirs $(OBJS) $(OLIBS) $(OIMAGES) $(OBINS) $(SPECIAL_MKTARGETS)
 
 .PHONY: sdk_extracted
 .PHONY: sdk_patched
 .PHONY: sdk_pruned
+.PHONY: toolchain
 
 sdk_extracted: $(TOP_DIR)/sdk/.extracted-$(SDK_BASE_VER)
 sdk_patched: sdk_extracted $(TOP_DIR)/sdk/.patched-$(SDK_VER)
 sdk_pruned: $(SDK_DIR_DEPENDS) $(TOP_DIR)/sdk/.pruned-$(SDK_VER)
 
+ifeq ($(OS),Windows_NT)
+toolchain:
+else
+toolchain: $(TOP_DIR)/tools/toolchains/$(TOOLCHAIN_VERSION)/esp8266/bin/xtensa-lx106-elf-gcc
+
+$(TOP_DIR)/tools/toolchains/$(TOOLCHAIN_VERSION)/esp8266/bin/xtensa-lx106-elf-gcc: $(TOP_DIR)/tools/toolchains/toolchain-esp8266-$(TOOLCHAIN_VERSION).tar.xz
+	mkdir -p $(TOP_DIR)/tools/toolchains/$(TOOLCHAIN_VERSION)/
+	tar -xJf $< -C $(TOP_DIR)/tools/toolchains/$(TOOLCHAIN_VERSION)/
+	touch $@
+
+$(TOP_DIR)/tools/toolchains/toolchain-esp8266-$(TOOLCHAIN_VERSION).tar.xz:
```

Comment:
Actually, could I get you to download this into the top-level `cache/` directory instead, right next to the SDK zip? That way one only has to `rm cache/*` to ensure one gets a fresh copy of everything downloaded.

Reply:
Sure, didn't know that one existed.